adoptopenjdk-api-v3-frontend/src/main/kotlin/net/adoptopenjdk/api/v3/routes/AssetsResource.kt

```diff
@@ -167,6 +148,89 @@ class AssetsResource {
         return getPage(pageSize, page, releases)
     }
 
+    @GET
+    @Path("/release_name/{vendor}/{release_name}")
+    @Operation(
+        summary = "Returns release information",
+        description = "List of releases with the given release name"
+    )
+    @APIResponses(
+        value = [
+            APIResponse(
+                responseCode = "200", description = "Release with the given vendor and name"
+            ),
+            APIResponse(responseCode = "400", description = "bad input parameter"),
+            APIResponse(responseCode = "404", description = "no releases match the request"),
+            APIResponse(responseCode = "500", description = "multiple releases match the request")
+        ]
+    )
+    fun get(
+        @Parameter(name = "vendor", description = OpenApiDocs.VENDOR, required = false)
+        @PathParam("vendor")
+        vendor: Vendor?,
+
+        @Parameter(name = "release_name", description = "Name of the release i.e ", required = true)
+        @PathParam("release_name")
+        releaseName: String?,
+
+        @Parameter(name = "os", description = "Operating System", required = false)
+        @QueryParam("os")
+        os: OperatingSystem?,
+
+        @Parameter(name = "architecture", description = "Architecture", required = false)
+        @QueryParam("architecture")
+        arch: Architecture?,
+
+        @Parameter(name = "image_type", description = "Image Type", required = false)
+        @QueryParam("image_type")
+        image_type: ImageType?,
+
+        @Parameter(name = "jvm_impl", description = "JVM Implementation", required = false)
+        @QueryParam("jvm_impl")
+        jvm_impl: JvmImpl?,
+
+        @Parameter(name = "heap_size", description = "Heap Size", required = false)
+        @QueryParam("heap_size")
+        heap_size: HeapSize?,
+
+        @Parameter(name = "project", description = "Project", required = false)
+        @QueryParam("project")
+        project: Project?
+    ): Release {
+        if (releaseName == null || releaseName.trim().isEmpty()) {
+            throw BadRequestException("Must provide a releaseName")
```

Comment:
do we want tests to cover the Bad Request scenarios?

Reply:
sure can do